SpatialGDK/Source/SpatialGDK/Private/LoadBalancing/WorkingSetSystem.cpp

```diff
@@ -0,0 +1,207 @@
+// Copyright (c) Improbable Worlds Ltd, All Rights Reserved
+
+#include "LoadBalancing/WorkingSetSystem.h"
+
+#include "Algo/AllOf.h"
+#include "Algo/Copy.h"
+
+#include "Interop/WorkingSetsCommon.h"
+#include "Schema/WorkingSet.h"
+#include "SpatialView/ComponentData.h"
+#include "SpatialView/EntityComponentTypes.h"
+#include "SpatialView/EntityDelta.h"
+#include "SpatialView/SubView.h"
+#include "SpatialView/ViewCoordinator.h"
+
+DEFINE_LOG_CATEGORY_STATIC(LogSpatialWorkingSetSystem, Log, All);
+
+namespace SpatialGDK
+{
+void FWorkingSetSystem::Advance(const FSubView& Subview)
+{
+	NewMembers.Empty();
+	EntitiesToRefresh.Empty();
+	NewMarkers.Empty();
+	FreedEntities.Empty();
+
+	for (const EntityDelta& Delta : Subview.GetViewDelta().EntityDeltas)
```

Comment:
We need to iterate twice - first process the removes from sets, then adds to sets.